src/emqx_plugrel.erl

```diff
@@ -5,31 +5,121 @@
 -define(LOG(LEVEL, FORMAT, ARGS),
         rebar_api:LEVEL("[emqx_plugrel] " ++ FORMAT, ARGS)).
 
--define(supported_pre_vsns(_CurrVsn), <<".*">>).
-
--define(PROVIDER, zip).
--define(DEPS, [release]).
-
 -spec init(rebar_state:t()) -> {ok, rebar_state:t()}.
 init(State) ->
     Provider = providers:create([
             {namespace, emqx_plugrel},
-            {name, ?PROVIDER},            % The 'user friendly' name of the task
-            {module, ?MODULE},            % The module implementation of the task
-            {bare, true},                 % The task can be run by the user, always true
-            {deps, ?DEPS},                % The list of dependencies
-            {example, "rebar3 emqx_plugrel zip"}, % How to use the plugin
-            {opts, []},                   % list of options understood by the plugin
+            {name, tar}, % The 'user friendly' name of the task
+            {module, ?MODULE}, % The module implementation of the task
+            {bare, true}, % The task can be run by the user, always true
+            {deps, [{default, release}]},  % The list of dependencies
+            {example, "rebar3 emqx_plugrel tar"}, % How to use the plugin
+            {opts, [emqx_plugrel]}, % list of options understood by the plugin
             {short_desc, "EMQ X plugin zip package"},
             {desc, "A rebar3 plugin that helps to release a zip package for EMQ X plugin"}
     ]),
     {ok, rebar_state:add_provider(State, Provider)}.
 
 do(State) ->
-    io:format(user, "~p\n", [State]),
+    Opts = rebar_state:opts(State),
+    Relx = rebar_opts:get(Opts, relx),
+    PluginInfo = rebar_opts:get(Opts, emqx_plugrel),
+    case lists:keyfind(release, 1, Relx) of
+        {release, {Name, Version}, Apps} ->
+            Info = collect_info(PluginInfo, Name, Version, Apps, State),
+            ok = make_tar(Info, State);
+        false ->
+            ?LOG(error, "relx_config_not_found", []),
+            error(relx_config_not_found)
+    end,
     {ok, State}.
 
 -spec format_error(any()) ->  iolist().
 format_error(Reason) ->
     io_lib:format("~p", [Reason]).
 
+collect_info(PluginInfo, Name, Version, Apps, State) ->
+    AppsWithVsn = lists:map(fun(App) -> resolve_vsn(App, State) end, Apps),
+    Info = info_map(PluginInfo),
+    MoreInfo = #{ name => bin(atom_to_list(Name))
+                , rel_vsn => bin(Version)
+                , rel_apps => AppsWithVsn
+                , build_time => now_time()
```

Comment:
Maybe use something deterministic for more reproducible builds, like commit hash?